ext/rp/timer.h

```diff
@@ -0,0 +1,34 @@
+// Subset of header "timer.h" from pico sdk
+
+/*
+ * Copyright (c) 2020 Raspberry Pi (Trading) Ltd.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+#ifndef _HARDWARE_TIMER_H
+#define _HARDWARE_TIMER_H
+
+#include "pico.h"
+#include "hardware/structs/timer.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/*! \brief Return a 32 bit timestamp value in microseconds
+*  \ingroup hardware_timer
+*
+* Returns the low 32 bits of the hardware timer.
+* \note This value wraps roughly every 1 hour 11 minutes and 35 seconds.
+*
+* \return the 32 bit timestamp
+*/
+static inline uint32_t time_us_32(void) {
+    return timer_hw->timerawl;
```

Comment:
Why not `modm::PreciseClock::now()`? Does TinyUSB configure the timer itself?

Reply:
I did not feel comfortable to change the code because I don't understand enough about the RP2040 platform to know if the modm timer does the right thing or not.

That `timer_hw` is some alternative platform specific microsecond timer. It was also mentioned in #907. TinyUSB does not configure it, but probably the platform initialization code does somehow. I was not able to figure out yet if it works with modm out of the box or not.

Reply:
Oh, then I should really fix that issue!

Reply:
That timer is actually only used if [TUD_OPT_RP2040_USB_DEVICE_UFRAME_FIX](https://github.com/hathach/tinyusb/blob/28817a715024c82b78f9b9d19f37a72771870759/src/portable/raspberrypi/rp2040/rp2040_usb.c#L381) is set. It enables a workaround for a silicon errata which is only relevant if the RP2040 is plugged into a RPi4. The issue does not occur with any other hub or host controller. It is quite expensive and reduces bulk transfer bandwidth by 20%. We could simply decide to not support the workaround for now if it's too much work.

Reply:
Yes, let's fix it later.
I want to use the hardware timer anyways to fix #907, just not right now.

Reply:
The "timer.h" header is now empty and the workaround is disabled. It has to be there because tinyusb always includes it, even when it is not used.